### PR TITLE
Fixed: Menu item is displaying as selected(#277)

### PR DESCRIPTION
**Pull request by shanmukhdutt**
### Related Issues
 

 Closes #277

 ### Short Description and Why It's Useful
 

 ### Screenshots of Visual Changes before/after 
 Before 
 
Screenshot from 2024-01-09 11-22-05

After

Screenshot from 2024-01-09 11-18-11

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [ ] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/components/Menu.vue [hunk 1]
```diff
@@ -95,29 +94,51 @@
...
+          url: "/threshold-updates",
+          childRoutes: ['/threshold-updates'],
+          iosIcon: pulseOutline,
+          mdicon: pulseOutline
+        },
+        {
+          title: "Settings",
+          url: "/Settings",
+          iosIcon: settingsOutline,
+          mdIcon: settingsOutline
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:126]: Add support for permission check, as we have support to check for permission for the pages

File: src/components/Menu.vue [hunk 2]
```diff
@@ -95,29 +94,51 @@
...
+        }
+      ];
+      const selectedIndex = computed(() => {
+      const path = router.currentRoute.value.path
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:130]: Improve indentation

File: src/components/Menu.vue [hunk 3]
```diff
@@ -7,18 +7,17 @@
...
-          <ion-icon :icon="pulseOutline" slot="start" />
-          <ion-label>{{ $t("Rule Pipeline") }}</ion-label>
-        </ion-item>
-        <ion-item button @click="closeMenu(); router.push('/settings')" class="hydrated" :class="{ selected: isSelected('/settings')}">
-          <ion-icon :icon="settingsOutline" slot="start" />
-          <ion-label>{{ $t("Settings") }}</ion-label>
-        </ion-item>
+        <ion-menu-toggle auto-hide="false" v-for="(p,i) in appPages" :key="i">
+          <ion-item 
+          button
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:12]: Improve indentation

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/Menu.vue [hunk 4]
```diff
@@ -95,26 +95,61 @@
...
+        appPages,
+        hasPermission,
         router,
         pulseOutline,
-        hasPermission,
         optionsOutline,
         settingsOutline,
-        store
+        store,
+        selectedIndex
```
Comment (amansinghbais) [inline src/components/Menu.vue:152]: arrange following in alphabetical order.

File: src/components/Menu.vue [hunk 5]
```diff
@@ -7,18 +7,17 @@
...
-        </ion-item>
-        <ion-item v-if="hasPermission('APP_THRESHOLD_UPDATES_VIEW')" button @click="closeMenu(); router.push('/threshold-updates')">
-          <ion-icon :icon="pulseOutline" slot="start" />
-          <ion-label>{{ $t("Rule Pipeline") }}</ion-label>
-        </ion-item>
-        <ion-item button @click="closeMenu(); router.push('/settings')">
-          <ion-icon :icon="settingsOutline" slot="start" />
-          <ion-label>{{ $t("Settings") }}</ion-label>
-        </ion-item>
+        <ion-menu-toggle auto-hide="false" v-for="(p,i) in getValidMenuItems(appPages)" :key="i">
```
Comment (amansinghbais) [inline src/components/Menu.vue:10]: The variables in v-for needs to be explanatory. Don't use character such as p and i.

File: src/components/Menu.vue [hunk 6]
```diff
@@ -95,26 +95,61 @@
...
+          url: "/threshold-updates",
+          iosIcon: pulseOutline,
+          mdIcon: pulseOutline,
+          meta: {
+            permissionId: "APP_THRESHOLD_UPDATES_VIEW"
+          }
+        },
+        {
+          title: "Settings",
+          url: "/Settings",
```
Comment (ymaheshwari1) [inline src/components/Menu.vue:133]:
```suggestion
          url: "/settings",
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**